## Wine Quality Dataset Analysis

Este análisis explora las características químicas de distintos vinos y su relación con la calidad evaluada por expertos.

El objetivo principal es comprender la estructura del dataset mediante técnicas de análisis exploratorio y métodos de aprendizaje no supervisado.

Se aplicarán tres enfoques principales:

1. Análisis exploratorio de datos (EDA)
2. Reducción de dimensionalidad mediante PCA
3. Agrupamiento de observaciones mediante K-Means

Estas técnicas permiten identificar patrones ocultos, relaciones entre variables y posibles agrupaciones naturales dentro del dataset.


------------

## Carga del dataset e inspección inicial

En esta sección se carga el dataset de calidad del vino y se revisa su estructura general.

El objetivo es confirmar que el archivo fue leído correctamente y verificar:

- número de filas y columnas,
- nombres de variables,
- valores faltantes,
- duplicados.

Esta revisión es importante antes de aplicar cualquier análisis estadístico o técnica de aprendizaje no supervisado.

In [1]:
from src.paquete_analisis_vino import AnalisisVino

analisis = AnalisisVino("data/raw/winequality-red.csv")

print(analisis.resumen_datos())

      métrica  valor
0     n_filas   1599
1  n_columnas     12
2  duplicados    240


**Inspección inicial**

El dataset contiene 1599 observaciones y 12 variables.
No se detectan valores nulos.

Existen 240 registros duplicados, los cuales se mantienen para conservar la estructura original del dataset.

### Estadísticas descriptivas

In [3]:
analisis.estadisticas()

,count,mean,std,min,25%,50%,75%,max
fixed acidity,1599.0,8.319637,1.741096,4.60000,7.1000,7.90000,9.200000,15.90000
volatile acidity,1599.0,0.527821,0.179060,0.12000,0.3900,0.52000,0.640000,1.58000
citric acid,1599.0,0.270976,0.194801,0.00000,0.0900,0.26000,0.420000,1.00000
residual sugar,1599.0,2.538806,1.409928,0.90000,1.9000,2.20000,2.600000,15.50000
chlorides,1599.0,0.087467,0.047065,0.01200,0.0700,0.07900,0.090000,0.61100
free sulfur dioxide,1599.0,15.874922,10.460157,1.00000,7.0000,14.00000,21.000000,72.00000
total sulfur dioxide,1599.0,46.467792,32.895324,6.00000,22.0000,38.00000,62.000000,289.00000
density,1599.0,0.996747,0.001887,0.99007,0.9956,0.99675,0.997835,1.00369
pH,1599.0,3.311113,0.154386,2.74000,3.2100,3.31000,3.400000,4.01000
sulphates,1599.0,0.658149,0.169507,0.33000,0.5500,0.62000,0.730000,2.00000


**Observación**

Las variables presentan escalas distintas, por lo que será necesario aplicar estandarización antes de los métodos de clustering.